src/js/abstract/base-component.js

```diff
@@ -1,121 +1,22 @@
-import nanomorph from './component-morph';
-import { isSameNodeOnce, clearIsSameNode } from './is-same-node-once';
-import getAttribute from '../get-attribute';
-import toProp from '../to-prop';
-import { publish, subscribe } from '../pubsub';
-import debounce from '../debounce';
-import camelize from '../camelize';
-import dasherize from '../dasherize';
-import maybe from '../maybe';
-import PropertyExistsException from './property-exists-exception';
+import getId from './utils/get-id';
+import lifecycleLogger from './utils/lifecycle-logger';
 
-const THROWED_ERROR = 'throwed';
-const PROPERTY_WHITELIST = ['title', 'checked', 'disabled'];
-const ids = {};
-const getId = (nodeName) => {
-  if (!(nodeName in ids)) {
-    ids[nodeName] = 0;
-  }
-
-  return ++ids[nodeName]; // eslint-disable-line no-plusplus
-};
-const lifecycleLogger = maybe((...args) => console.log(...args))()(true);
+import withContext from './hocs/with-context';
+import withMonkeyPatches from './hocs/with-monkey-patches';
+import withUpdate from './hocs/with-update';
+import withRender from './hocs/with-render';
+import withStyles from './hocs/with-styles';
 
 /**
  * Base class {BaseComponent}. This class checks if a template is set in the custom element
  * and if yes appends it. It also appends custom styles to the top of the dom tree.
- *
- * **Light DOM**
- * The light DOM are the provided children from the users of your component (light meaning easy to digest).
- *
- * ```html
- * <axa-example>
- *   <div>This is some light DOM for axa-example</div>
- * </axa-example>
- * ```
- *
- * **Local DOM**
- * The local DOM is the DOM tree rendered by the component itself (in our case provided by `template`).
- *
- * ```js
- * export default function(props, childrenFragment) {
- *   return nanohtml`<article>
- *     ${childrenFragment} <!-- light DOM injection point -->
- *   </article>`;
- * }
- * ```
- *
- * **Flattened DOM**
- * The flattened DOM is the final product where the user's light DOM is injected into the Components local DOM.
- *
- * ```html
- * <axa-example>
- *   <article>
- *     <div>This is some light DOM for axa-example</div> <!-- light DOM injection point -->
- *   </article>
- * </axa-example>
- * ```
  */
-export default class BaseComponent extends HTMLElement {
+class BaseComponent extends HTMLElement {
   constructor(styles = '', template) {
```

Comment:
@LucaMele 
I would like to change the constructor to accept only a single arg of type `object`.
That we can forward this hash to each `super({})` call, and releated hoc can just destruct what they need 👍